src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs

```diff
@@ -1473,6 +1507,29 @@ public void ProcessFileChange(string projectSystemFilePath, string workspaceFile
                 return solution;
             }
 
+            internal Solution UpdateProjectDocumentsOrderForBatch(Solution oldSolution, ProjectId projectId)
+            {
+                var orderedFilePaths = _orderedFilePathsInBatch.ToImmutableArray();
+                if (orderedFilePaths.IsEmpty)
+                {
+                    return oldSolution;
+                }
+
+                _orderedFilePathsInBatch.Clear();
+
+                return UpdateProjectDocumentsOrder(oldSolution, projectId, orderedFilePaths);
+            }
+
+            private Solution UpdateProjectDocumentsOrder(Solution oldSolution, ProjectId projectId, IEnumerable<string> filePaths)
+            {
+                var documentIds =
+                    filePaths
+                        .Select(x => oldSolution.GetDocumentIdsWithFilePath(x).FirstOrDefault(id => id.ProjectId == projectId))
+                        .Where(x => x != null);
```

Comment:
how do we get the 'null' case?  woudln't that indicate a very bad bug? i.e. trying to specify an ordering of files, but the file isn't even in the project that is being ordered?

Reply:
It is possible to get a null case if you pass a file path that doesn't exist. It just excludes it from the order list.

Reply:
What is the case where the host passes a file that doesn't exist along?  How could the files have been ordered in the first place if the file doesn't exist?